### PR TITLE
Refactor and enhance generated Object tests

### PR DESCRIPTION
**Pull request by LittleFox94**
### Description

This PR refactors the generated Object tests to have most logic in a normal package which is imported by the generated code, rather than having the logic in the template string - making the generated tests much better to maintain.

Also adds two additional checks:
* test if the implemented hooks work fine with incomplete tests (needed for 100% test coverage, easy to test correctly with generated tests)
* test if the Object has an `identifier` field
  + we previously only had runtime errors for this

### Checklist

* [ ] added release notes to `Unreleased` section in [CHANGELOG.md](CHANGELOG.md), if user facing change

### References

* #84 

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

**Comment by LittleFox94** [comment]
whoops, right, didn't implement the checks for all hooks, yet x)

File: pkg/utils/test/object.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+			"ResponseFilterHook": func(ctx context.Context) error {
+				rec := httptest.NewRecorder()
+				rec.WriteHeader(200)
+				_, _ = rec.WriteString(`{}`)
+				_, err := o.(types.ResponseFilterHook).FilterAPIResponse(ctx, rec.Result())
+				return err
+			},
+		}
+
+		if errorCheck, ok := supportedHooks[name]; ok {
```
Comment (kstiehl) [inline pkg/utils/test/object.go:64]: what if `hook` is not in the list of supportedHooks. Do we just not test it then? Or should we at least test whether the hook is actually implemented.

File: pkg/utils/test/object.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+// * calling the hook function with incomplete contexts gives none or the correct error (meaning the error is handled correctly)
+//
+func ObjectTests(o types.Object, hooks ...interface{}) {
+	ginkgo.It("has an identifier", func() {
+		_, err := api.GetObjectIdentifier(o, false)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	})
+
+	for _, hook := range hooks {
+		name := reflect.TypeOf(hook).Elem().Name()
```
Comment (kstiehl) [inline pkg/utils/test/object.go:32]: I would move this line a few lines down. To initialize the variable shortly before it is used. makes it a little bit more readable

Comment (LittleFox94) [reply pkg/utils/test/object.go:32]: ack

File: pkg/utils/test/object.go (hunk 1 above)
Comment (LittleFox94) [reply pkg/utils/test/object.go:64]: good catch, had code organized differently before where this made sense - will change

**Review by kstiehl — APPROVED** [review]
LGTM

**Pull request merged**